Source/Mosa.Compiler.MosaTypeSystem.CLR/Metadata/ClrMetadataCache.cs

```diff
@@ -1,50 +1,49 @@
 ﻿// Copyright (c) MOSA Project. Licensed under the New BSD License.
 
-using System;
-using System.Collections.Generic;
 using dnlib.DotNet;
 using Mosa.Compiler.Common.Exceptions;
+using Mosa.Compiler.MosaTypeSystem.CLR.Utils;
 
-namespace Mosa.Compiler.MosaTypeSystem.Metadata
+namespace Mosa.Compiler.MosaTypeSystem.CLR.Metadata
 {
-	public class MetadataCache
+	public class ClrMetadataCache
 	{
-		public Dictionary<string, MosaModule> Modules { get; }
+		public Dictionary<string, MosaModule?> Modules { get; }
 
 		public MosaModule LinkerModule { get; }
 
 		public MosaType DefaultLinkerType { get; }
 
-		private readonly Dictionary<ScopedToken, MosaType> typeLookup = new Dictionary<ScopedToken, MosaType>();
-		private readonly Dictionary<ScopedToken, MosaMethod> methodLookup = new Dictionary<ScopedToken, MosaMethod>();
-		private readonly Dictionary<ScopedToken, MosaField> fieldLookup = new Dictionary<ScopedToken, MosaField>();
-		private readonly Dictionary<ScopedToken, MosaProperty> propertyLookup = new Dictionary<ScopedToken, MosaProperty>();
+		private readonly Dictionary<ScopedToken, MosaType> typeLookup = new();
+		private readonly Dictionary<ScopedToken, MosaMethod> methodLookup = new();
+		private readonly Dictionary<ScopedToken, MosaField> fieldLookup = new();
+		private readonly Dictionary<ScopedToken, MosaProperty> propertyLookup = new();
 
 		private uint stringIdCounter;
-		internal Dictionary<string, uint> stringHeapLookup = new Dictionary<string, uint>(StringComparer.Ordinal);
-		internal Dictionary<uint, string> stringHeapLookup2 = new Dictionary<uint, string>();
+		private readonly Dictionary<string, uint> stringHeapLookup = new(StringComparer.Ordinal);
+		private readonly Dictionary<uint, string> stringHeapLookup2 = new();
 
-		public MetadataCache()
+		public ClrMetadataCache()
 		{
-			Modules = new Dictionary<string, MosaModule>();
+			Modules = new Dictionary<string, MosaModule?>();
 		}
 
-		public void AddModule(MosaModule module)
+		public void AddModule(MosaModule? module)
 		{
-			Modules.Add(module.Name, module);
+			Modules.Add(module?.Name, module);
```

Comment:
Fx warnings

Reply:
Unfortunately, I can't really fix this because the key of a Dictionary should never be null.